docs/proposals/image-signing.md

```diff
@@ -0,0 +1,129 @@
+# Image Signing
+
+## Abstract
+
+This proposal describes the process of verifying the signatures of Docker Images and
+enforce enforce policy that will prevent the image from running in the cluster if the
+image signature is not trusted.
+
+## Use-cases
+
+* As an admin, I want to be able to add trusted GPG public keys to my cluster and determine through policy
+  the keys that are required to run certain images
+* As an user, I want to be able to see what images are signed and if the signature is trusted
+
+## Overview
+
+### Signing Images with Skopeo
+
+The current signing mechanism for Docker image is to use the
+[Skopeo](https://github.com/projectatomic/skopeo/blob/master/docs/skopeo.1.md) tool which
+will sign the Docker image locally, using the GPG key and attach the signature to the
+Docker Image. The Docker image is then pushed to a Docker Registry with enabled signature
+storage.
+
+Signature verification [policy
+files](https://github.com/containers/image/blob/master/docs/policy.json.md) are used to
+define policy, e.g. trusted keys, applicable when deciding whether to accept an image, or
+individual signatures of that image, as valid:
+
+```json
+{
+  "default": [{"type": "reject"}],
+  "transports": {
+    "atomic": {
+      "172.30.62.231:5000": [
+      {
+        "type": "signedBy",
+        "keyType": "GPGKeys",
+        "keyPath": "/keys/public.gpg"
+      }
+      ]
+    },
+    "docker": {
+      "docker.io/openshift": [{"type": "insecureAcceptAnything"}]
+    }
+  }
+}
+```
+
+In this example, the default policy is set to reject all Docker images by default and
+whitelist only images that are pushed in the OpenShift Integrated Registry and signed
+using the public GPG key.
+It also whitelists all Docker images from the OpenShift DockerHub repository.
+
+To sign an Docker image, you have to execute the following command:
+
+```console
+$ skopeo --tls-verify=false copy --sign-by user@example.com docker://docker.io/myrepo/image:latest \
+  atomic:172.30.74.246:5000/mynamespce/image:latest
+```
+
+When this command is executed, skopeo will ask you to enter the password the
+*user@example.com* GPG key, sign the local image and push the signed image into OpenShift
+Integrated Registry.
+
+### Kubernetes `ImagePolicyWebhook` Admission Controller
+
+Kubernetes provides the
```

Comment:
Update section to discuss both the upstream policy webhook and the existing OpenShift ImagePolicy engine.  We have a choice here in terms of implementation. We can enable their admission controller or embed it into our existing image policy admission controller.